src/storage/blocksstable/ob_macro_block.h

```diff
@@ -53,9 +53,9 @@ struct ObDataStoreDesc {
   bool is_major_;
 
   char compressor_name_[common::OB_MAX_HEADER_COMPRESSOR_NAME_LENGTH];
-  uint64_t column_ids_[common::OB_MAX_COLUMN_NUMBER];
-  common::ObObjMeta column_types_[common::OB_MAX_COLUMN_NUMBER];
-  common::ObOrderType column_orders_[common::OB_MAX_COLUMN_NUMBER];
+  storage::ObDynamicBuffer<uint64_t> column_ids_;
+  storage::ObDynamicBuffer<ObObjMeta> column_types_;
+  storage::ObDynamicBuffer<ObOrderType> column_orders_;
```

Comment:
in deconstructor allocator_ will be released before buffer?

Reply:
fixed.